### PR TITLE
#3068 Experience names are displayed incompletely (fix batch rendering)

### PR DESCRIPTION
**Pull request by LLGuru**

### CODE REVIEW

File: indra/llrender/llfontgl.cpp [hunk 1]
```diff
@@ -341,24 +349,35 @@ S32 LLFontGL::render(const LLWString &wstr, S32 begin_offset, F32 x, F32 y, cons
...
             bitmap_entry.first == EFontGlyphType::Grayscale ? text_color
                                                             : emoji_color;
+        // This function emplaces data to arrays vertices, uvs, colors
+        // and increases glyph_count on 1, 2 or 6 depending on shadow
         drawGlyph(glyph_count, vertices, uvs, colors, screen_rect, uv_rect,
                   col, style_to_add, shadow, drop_shadow_strength);
+        llassert(glyph_count <= GLYPH_BATCH_SIZE);
+        if (!glyph_per_char)
+        {
+            glyph_per_char = glyph_count;
```
Comment (akleshchev) [inline indra/llrender/llfontgl.cpp:365]: Shouldn't this be drawGlyph(glyph_per_char, ... and glyph_count += glyph_per_char;?

Comment (LLGuru) [reply indra/llrender/llfontgl.cpp:365]: Definitely No

Each call of `drawGlyph` increases `glyph_count` on the same value (1, 2 or 6, depending on `shadow`)
Initially `glyph_per_char` is set to 0, and only after the first call of `drawGlyph` it is reinitialized again
Using this value we can prognose the next size of `glyph_count` before the next call of `drawGlyph`

Comment (akleshchev) [reply indra/llrender/llfontgl.cpp:365]: Aren't you missing a reset in such case?

Comment (LLGuru) [reply indra/llrender/llfontgl.cpp:365]: Yes, the parameter `shadow` is the same for all characters in the provided string,
so the value of `glyph_per_char` is absolutely the same for the whole string fragment

File: indra/llrender/llfontgl.cpp [hunk 2]
```diff
@@ -295,30 +300,33 @@ S32 LLFontGL::render(const LLWString &wstr, S32 begin_offset, F32 x, F32 y, cons
...
         {
             LL_ERRS() << "Missing Glyph Info" << LL_ENDL;
             break;
         }
+
         // Per-glyph bitmap texture.
-        std::pair<EFontGlyphType, S32> next_bitmap_entry = fgi->mBitmapEntry;
-        if (next_bitmap_entry != bitmap_entry)
+        const std::pair<EFontGlyphType, S32>& bitmap_entry = fgi->mBitmapEntry;
+        if (wch != last_wch || i == 0)
```
Comment (akleshchev) [inline indra/llrender/llfontgl.cpp:324]: last_wch doesn't get set anywhere?

**Pull request merged**